File: src/courses/data/catalog/02-dsa/dsa-core/implementations/javascript/hashing.js
/**
 * JavaScript Implementation: Hashing (Hash Table & Map)
 * Invariant: Amortized O(1) Lookup/Insert/Delete
 */
export class HashingHashTableMap {
  constructor() {
    this.items = [];
  }

  // TODO: Implement verified invariant
  insert(val) {
    this.items.push(val);
  }

  get size() {
    return this.items.length;
  }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/hashing.java
/**
 * Java Implementation: Hashing (Hash Table & Map)
 * Invariant: Amortized O(1) Lookup/Insert/Delete
 */
package seed.dsa.core;

import java.util.*;

public class HashingHashTableMap<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
